### ISSUE TITLE
[OH3] [JSR223] Rule folder watcher does not detect file create/modify/delete

### ISSUE THREAD

**Issue by CrazyIvan359** [openhab/openhab-core#1820]
The JSR223 rule/script folder watcher pointed at `{OH_CONF}/automation/jsr223` (possibly just `{OH_CONF}/automation`) does not detect any file add/modify/delete events. I have tried with and without symlinks, with and without nested folders, nothing is detected. The only way to load new or modified scripts is by restarting OH.

I am running OH3 snapshot 2006, but had the same behaviour on 2005. I have only tested this with JS/ECMA scripts.

I went digging in the code and found that it is set to follow symlinks and descend into subfolders, so that would not appear to be the issue.

Possibly related to #1653 ?

**Comment by CrazyIvan359** [comment]
#1859 is possibly related

**Comment by kaikreuzer** [comment]
I cannot reproduce this on the latest snapshot - I'd assume that this was fixed by https://github.com/openhab/openhab-core/pull/1914.

Feel free to re-open, if you can still reproduce it.

**Comment by CrazyIvan359** [comment]
Thanks Kai! I will test that today or tomorrow and report back if it is not working.

**Comment by CrazyIvan359** [comment]
@kai sorry it took me so long to get back to you. This is only partially working on S2100, I think. It works for `conf/automation/jsr223/` and subdirs, but it does not follow symlinks. I dug into it before and I was pretty sure the rule engine folder watcher was configured to follow symlinks.

If the rule engine is changed to monitor `conf/rules` as well then it becomes a moot point, as that is where my symlink points. I did this because it seemed to only make sense to me for OH3, given that the DSL rules are now like any other and are part of the rule engine. If the DSL is just another SE Factory, why shouldn't all rules go in `conf/rules` now? Anyway, food for though, I have just moved my rules into `conf/automation/jsr223/*` for now.
